plugins/pur/core/src/main/java/org/pentaho/di/repository/pur/PurRepository.java

```diff
@@ -1492,13 +1497,17 @@ private LinkedHashMap<RepositoryObjectType, List<RepositoryFile>> getFilesByType
   public List<DatabaseMeta> readDatabases() throws KettleException {
     readWriteLock.readLock().lock();
     try {
+      boolean hasOsgiFolder = new File( SYSTEM_FOLDER ).exists();
       List<RepositoryFile> children = getAllFilesOfType( null, RepositoryObjectType.DATABASE, false );
       List<DatabaseMeta> dbMetas = new ArrayList<DatabaseMeta>();
 
       for ( RepositoryFile file : children ) {
         DataNode node;
         node = pur.getDataForRead( file.getId(), NodeRepositoryFileData.class ).getNode();
-
+        if ( !hasOsgiFolder && StringUtils.equals( node.getProperty( "TYPE" ).getString(), "KettleThin" ) ) {
+          log.logDetailed( "Unable to find database {" + file.getName() + "}" );
+          continue;
+        }
```

Comment:
**Post Merge Comment**
Imo, this would better be done in the error handler, where the severity of the error would be evaluated, and a decision of logging and continuing or throwing would be made.

Also, I think that rather than hard-coding the `KettleThin` connection type, it would be better to recognize that, generally, a repository may store connections whose type the current PDI client/engine does not know of. This is expected, given the different flavors of PDI clients/engines which exist, and in an application following a plugin architecture.

In this case, the `KettleThin` connection type will not be used by any other feature installed in the slimmed-down PDI client/engine, and, if it were, that dependency would likely be manifested as a JAVA or OSGi dependency between plugins of the engine, and, when unsatisfied, would be caught at engine boot time. 

Execution of transformations/jobs which would require the specific connection type or connection instance to be defined (not sure this kind of dependency exists for this specific case) will not be possible in this PDI client/engine. That's an unavoidable condition for a pluggable application with documents. The only question is how and when these dependencies are detected and their unsatisfaction handled.

Concluding, the particular condition of a connection type being unknown — for any unknown connection type — should be detectable by either catching a specific exception class or inspecting the caught error. That specific condition would be logged and connection loading would be resumed.

Reply:
Hi, Thanks for reviewing and sharing the inputs.
Actually in DatabaseMeta exception is thrown if the plugin type is not available
https://github.com/pentaho/pentaho-kettle/blob/master/core/src/main/java/org/pentaho/di/core/database/DatabaseMeta.java#L592

are you suggesting to make change at DatabaseMeta to just log and remove this line ? If yes, we can revert the code changes for this PR